server/config.go

```diff
@@ -203,7 +203,15 @@ func ReadConfig(ctx *cli.Context) Config {
 		FallbackTargets:        ctx.StringSlice(FallbackTargets),
 		CacheTargets:           ctx.StringSlice(CacheTargets),
 	}
-	cfg.ClientConfig.WaitForFinalization = (cfg.EthConfirmationDepth < 0)
+	// the eigenda client can only wait for 0 confirmations or finality
+	// the da-proxy has a more fine-grained notion of confirmation depth
+	// we use -1 to let the da client wait for finality, and then need to set the confirmation depth
+	// for the da-proxy to 0 (because negative confirmation depth doesn't mean anything and leads to errors)
+	// TODO: should the eigenda-client implement this feature for us instead?
```

Comment:
wouldn't that also require the client to do the cert verifications as well?

Reply:
Let's chat about this! I don't understand why you are saying this (don't have a full model of how all of these things are tied together), but I created https://github.com/Layr-Labs/eigenda-proxy/issues/130 specifically to upstream bunch of these things to the da-client. Is that possible? Is there a reason we haven't done that?

Reply:
Merging. Let's continue discussion in that issue.